lib/mail/field.rb

```diff
@@ -232,12 +232,13 @@ def split(raw_field)
     #  that is immediately followed by WSP.  Each header field should be
     #  treated in its unfolded form for further syntactic and semantic
     #  evaluation.
-    def unfold(string)
-      string.gsub(/[\r\n \t]+/m, ' ')
+    def unfold(name, value)
+      regex = name == 'Subject' ? /\s*\r\n\s+/m : /[\r\n \t]+/m
```

Comment:
@kjg. That's quite possible, that the specs are wrong. But I wanted to cause as little disruption as possible.

An example it the 'To' header, in the spec right above the 2 I added:

```ruby
    it "should remove multiple spaces during unfolding a header" do
      header = Mail::Header.new("To: Mikel,\r\n   Lindsaar,     Bob")
      expect(header['To'].value).to eq 'Mikel, Lindsaar, Bob'
    end
```